datashader/glyphs.py

```diff
@@ -28,11 +29,45 @@ def validate(self, in_dshape):
         elif not isreal(in_dshape.measure[self.y]):
             raise ValueError('y must be real')
 
-    def _compute_x_bounds(self, df):
-        return df[self.x].min(), df[self.x].max()
+    @staticmethod
+    @ngjit
+    def _compute_x_bounds(xs):
+        minval = maxval = xs[0]
```

Comment:
Also, I'm on vacation this week - feel free to make additional changes to this PR as you see fit.